File: core/src/main/java/com/pluxity/station/dto/StationUpdateRequest.java
package com.pluxity.station.dto;

import com.pluxity.facility.dto.FacilityUpdateRequest;
import com.pluxity.facility.floor.dto.FloorRequest;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.List;

public record StationUpdateRequest(
        @Schema(requiredMode = Schema.RequiredMode.REQUIRED) FacilityUpdateRequest facility,
        @Schema(requiredMode = Schema.RequiredMode.REQUIRED) List<FloorRequest> floors,
        @Schema(requiredMode = Schema.RequiredMode.REQUIRED) StationUpdateInfo stationInfo) {}

File: core/src/main/kotlin/com/pluxity/station/dto/StationUpdateRequest.kt
package com.pluxity.station.dto

import com.pluxity.facility.dto.FacilityUpdateRequest
import com.pluxity.facility.floor.dto.FloorRequest
import io.swagger.v3.oas.annotations.media.Schema

data class StationUpdateRequest(
    @field:Schema(requiredMode = Schema.RequiredMode.REQUIRED)
    val facility: FacilityUpdateRequest,
    @field:Schema(requiredMode = Schema.RequiredMode.REQUIRED)
    val floors: List<FloorRequest> = emptyList(),
    @field:Schema(requiredMode = Schema.RequiredMode.REQUIRED)
    val stationInfo: StationUpdateInfo? = null,
)
